### PR TITLE
API 4.0.0

### PR DESCRIPTION
**Pull request by TheRealRyGuy**
- Updates gradle wrapper to 8.1.1, as well as gradle plugins
- Updates workflow versions, uses gradle validation 
- Uses proper submodule setup 
- Uses java 11 
- Uses API version 4.0.0

Note: Releasing will have to be altered with the shadow update, likely just to fix naming conventions. This will also require updating the release workflow

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -18,7 +19,7 @@ repositories {
 
 dependencies {
     compileOnly 'org.spigotmc:spigot-api:1.19.2-R0.1-SNAPSHOT'
-    implementation 'gs.mclo:api:3.0.1'
+    implementation project(path: ":mclogs-java", configuration: 'default')
```
Comment (JulianVennen) [inline build.gradle:22]: Please continue using maven central here, 4.0.1 is already synced

**Review by JulianVennen — CHANGES_REQUESTED** [review]

File: build.gradle [hunk 2]
```diff
@@ -1,8 +1,9 @@
 plugins {
     id 'java'
-    id 'com.github.johnrengelman.shadow' version '6.1.0'
-    id "com.modrinth.minotaur" version "2.+"
-    id "xyz.jpenilla.run-paper" version "2.0.1"
+    id 'com.github.johnrengelman.shadow' version '8.1.1'
+    id "com.modrinth.minotaur" version "2.8.1"
+    id "xyz.jpenilla.run-paper" version "2.1.0"
+    id "maven-publish"
```
Comment (JulianVennen) [inline build.gradle:6]: not sure why you added maven-publish

**Review by JulianVennen — CHANGES_REQUESTED** [review]

**Comment by JulianVennen** [comment]
I'll merge this tomorrow, I can also do these changes myself then. Once again, thanks for your contributions.

**Comment by TheRealRyGuy** [comment]
all done, and pulled some other things you did today from other repos :+1: didn't realize I missed so much, apologies

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -4,13 +4,16 @@ jobs:
...
+    - name: Set up JDK 11
       uses: actions/setup-java@v3
       with:
         distribution: temurin
-        java-version: 8
+        java-version: 11
     - name: Grant execute permission for gradlew
       run: chmod +x gradlew
+    - name: init submodule
+      run: git submodule init && git submodule update
```
Comment (JulianVennen) [inline .github/workflows/build.yml:17]: this is no longer required

File: settings.gradle [hunk 4]
```diff
@@ -1,7 +1,3 @@
...
-    repositories {
-        gradlePluginPortal()
-    }
-}
-
-rootProject.name = 'mclogs-bukkit'
\ No newline at end of file
+rootProject.name = 'mclogs-bukkit'
+include ':mclogs-java'
+project(':mclogs-java').projectDir = new File('./mclogs-java/')
```
Comment (JulianVennen) [inline settings.gradle:3]: this is also not required anymore

**Review by JulianVennen — CHANGES_REQUESTED** [review]

**Pull request merged**